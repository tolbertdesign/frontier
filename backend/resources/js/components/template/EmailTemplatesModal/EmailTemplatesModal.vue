<template>
  <AccordionModal
    :header="lang.emails_we_send"
    class="mx-4"
  >
    <template>
      <Accordion
        v-for="(template, index) in templates"
        :key="template.id"
        :open="template.isOpen"
        animation="zoom-out"
        @open="openAndCloseOthers(index)"
      >
        <template slot="title">
          <p class="text-sm sm:text-base md:text-lg text-left">
            <span class="font-semibold">{{ template.id }}. {{ template.name }}</span><br>
            {{ template.snippet }}
          </p>
        </template>
        <template>
          <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
            <p class="mb-4">
              {{ lang.sponsor_name }},
            </p>
            <p class="mb-4">
              {{
                parseLanguage(
                  lang.template,
                  {
                    event_name: program.event_name,
                    names: formattedNames,
                    is_are: isAre,
                    funds_raised_for: program.microsite.funds_raised_for,
                    unit_name: program.unit.name,
                    plural_unit_name: program.unit.name_plural,
                    pledge_goal: participant.profile.pledge_goal | currency,
                    unit_modifier: program.unit.modifier,
                  }
                )
              }}
            </p>
            <p class="mb-4">
              <a :href="template.link">
                {{ parseLanguage(lang.click_here_reach_for_greatness, { names: formattedNames }) }}
              </a>
            </p>
            <p class="mb-4">
              {{ lang.thanks_so_much }}
            </p>
            <p class="mb-4">
              - {{ user.first_name }}
            </p>
          </div>
        </template>
      </Accordion>
    </template>
    <template slot="footer">
      <AddContactsButtons />
    </template>
  </AccordionModal>
</template>

<script>
import AddContactsButtons from '@/components/template/AddContactsButtons'
import Accordion from '@/components/element/Accordion'
import AccordionModal from '@/components/template/AccordionModal'
import Blur from '@/mixins/Blur'

export default {
  name: 'EmailTemplatesModal',
  components: {
    AddContactsButtons,
    Accordion,
    AccordionModal,
  },
  mixins: [Blur],
  data () {
    return {
      EE_ENROLLMENT: 6,
      EE_DAY_BEFORE: 7,
      EE_DAY_AFTER: 8,
    }
  },
  computed: {
    lang () {
      return this.$store.state.lang
    },
    templates () {
      return [
        {
          id: 1,
          name: 'After Contact is Added',
          snippet: 'Subject: Help ' + this.formattedNames + ' in the ' + this.program.event_name,
          isOpen: false,
          link: this.enrollmentLink,
        },
        {
          id: 2,
          name: 'One Day Before The Event',
          snippet: 'Subject: Help ' + this.formattedNames + ' in the ' + this.program.event_name,
          isOpen: false,
          link: this.dayBeforeLink,
        },
        {
          id: 3,
          name: 'One Day Before Event Ends',
          snippet: 'Subject: Help ' + this.formattedNames + ' in the ' + this.program.event_name,
          isOpen: false,
          link: this.dayAfterLink,
        },
      ]
    },
    enrollmentLink () {
      let link = ''
      this.participant.special_urls.forEach(specialUrl => {
        if (specialUrl.referrer.id === this.EE_ENROLLMENT) {
          link = '/a/s/' + specialUrl.short_key
        }
      })
      return link
    },
    dayBeforeLink () {
      let link = ''
      this.participant.special_urls.forEach(specialUrl => {
        if (specialUrl.referrer.id === this.EE_DAY_BEFORE) {
          link = '/a/s/' + specialUrl.short_key
        }
      })
      return link
    },
    dayAfterLink () {
      let link = ''
      this.participant.special_urls.forEach(specialUrl => {
        if (specialUrl.referrer.id === this.EE_DAY_AFTER) {
          link = '/a/s/' + specialUrl.short_key
        }
      })
      return link
    },
    participants () {
      return this.program.participants
    },
    participant () {
      return this.participants.find(participant => {
        return participant.id === parseInt(this.$route.params.participantUserId)
      })
    },
    user () {
      return this.$store.state.User
    },
    program () {
      return this.user.programs.find(program => {
        return program.participants.find(participant => participant.id === parseInt(this.$route.params.participantUserId)) !== undefined
      })
    },
    participantLength () {
      return this.participants.length
    },
    formattedNames () {
      let names = ''
      if (this.participantLength > 2) {
        this.participants.forEach((participant, index) => {
          // Check if is last participant
          if (index + 1 === this.participantLength) {
            names += 'and ' + participant.first_name
          } else {
            names += participant.first_name + ', '
          }
        })
      } else if (this.participantLength === 2) {
        names = this.participants[0].first_name + ' & ' + this.participants[1].first_name
      } else if (this.participantLength === 1) {
        names = this.participants[0].first_name
      }

      return names
    },
    isAre () {
      if (this.participantLength > 1) {
        return 'are'
      }

      return 'is'
    },
  },
  methods: {
    openAndCloseOthers (index) {
      this.templates.forEach(template => {
        template.isOpen = false
      })
      this.templates[index].isOpen = true
    },
    closeModal () {
      this.$emit('close')
      this.unBlur()
    },
  },
}
</script>
