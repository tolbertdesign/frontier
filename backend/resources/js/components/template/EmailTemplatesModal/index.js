import EmailTemplatesModal from './EmailTemplatesModal'

export default EmailTemplatesModal
