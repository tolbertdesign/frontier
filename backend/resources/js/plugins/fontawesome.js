import Vue from 'vue'

import { library, dom } from '@fortawesome/fontawesome-svg-core'

// Brands (fab)
import { faFacebook, faFacebookF, faFacebookSquare, faTwitter } from '@fortawesome/free-brands-svg-icons'

// Regular (far)
import {
  faAddressCard,
  faArrowCircleRight,
  faBars as farBars,
  faCircle,
  faCommentAltLines,
  faEdit,
  faEnvelope,
  faExternalLinkSquare,
  faInfoCircle,
  faLock,
  faSignIn,
  faSignOut,
  faSquare,
  faTimes as farTimes,
  faUserEdit as farUserEdit,
} from '@fortawesome/pro-regular-svg-icons'

// Light (fal)
import { faBars, faTimes, faUserEdit } from '@fortawesome/pro-light-svg-icons'

// Solid Style Icons (fas)
import {
  faBars as fasBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle as fasCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faEdit as fasEdit,
  faEnvelope as fasEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faEye,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock as fasLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  faShare,
  faShareSquare,
  faSortDown,
  faSpinner,
  faStrikethrough,
  faTimes as fasTimes,
  faTrashAlt,
  faUnderline,
  faUndo,
  faUser,
  faUsers,
  faUserPlus,
} from '@fortawesome/free-solid-svg-icons'

import {
  FontAwesomeIcon,
  FontAwesomeLayers,
  FontAwesomeLayersText,
} from '@fortawesome/vue-fontawesome'

library.add(
  faAddressCard,
  faArrowCircleRight,
  faBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faCommentAltLines,
  faEdit,
  fasEdit,
  faEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faExternalLinkSquare,
  faEye,
  faFacebook,
  faFacebookF,
  faFacebookSquare,
  faInfoCircle,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  farBars,
  farTimes,
  fasBars,
  fasCircle,
  fasEnvelope,
  faShare,
  faShareSquare,
  faSignIn,
  faSignOut,
  fasLock,
  faSortDown,
  faSpinner,
  faSquare,
  fasTimes,
  faUsers,
  faStrikethrough,
  faTimes,
  faTrashAlt,
  faTwitter,
  faUnderline,
  faUndo,
  faUser,
  faUserEdit,
  faUserPlus,
  farUserEdit,
)

dom.watch()

Vue.component('FontAwesomeIcon', FontAwesomeIcon)
Vue.component('FontAwesomeLayers', FontAwesomeLayers)
Vue.component('FontAwesomeLayersText', FontAwesomeLayersText)
