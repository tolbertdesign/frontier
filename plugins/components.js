import Vue from 'vue'

/*
 ** Import all components from `base/` and `icon/` directories
 ** Learn more on require.context() on https://webpack.js.org/guides/dependency-management/#requirecontext
 */
function importComponents(r) {
  r.keys().forEach(filename => {
    let Component = r(filename)

    Component = Component.default || Component
    Component.name =
      Component.name || filename.replace(/^.+\//, '').replace(/\.\w+$/, '')
    Vue.component(Component.name, Component)
  })
}

importComponents(require.context('@/components/core', true, /\.(vue|js)$/))
importComponents(require.context('@/components/icons', true, /\.(vue|js)$/))
importComponents(require.context('@/components/examples', true, /\.(vue|js)$/))
