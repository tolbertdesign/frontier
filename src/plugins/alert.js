import Vue from 'vue'

import { VAlert } from 'vuetensils'

Vue.component('VAlert', VAlert)

// VueEssentials
