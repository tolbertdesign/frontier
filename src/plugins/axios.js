import Vue from 'vue'
import axios from 'axios'
import Swal from 'sweetalert2'
import store from '@/vuex/store'
import router from '@/router'
import i18n from '@/plugins/i18n'

const baseURL = 'http://tolbert.test'

const instance = axios.create({
  baseURL: baseURL,
  headers: {
    'X-Requested-With': 'XMLHttpRequest',
  },
})
instance.defaults.withCredentials = true

// Request interceptor
instance.interceptors.request.use(request => {
  const token = store.getters['auth/token']
  if (token) {
    request.headers.common.Authorization = `Bearer ${token}`
  }

  const locale = store.getters['lang/locale']
  if (locale) {
    request.headers.common['Accept-Language'] = locale
  }

  // request.headers['X-Socket-Id'] = Echo.socketId()

  return request
})

// Response interceptor
instance.interceptors.response.use(response => response, error => {
  const { status } = error.response

  if (status >= 500) {
    Swal.fire({
      type: 'error',
      title: i18n.t('error_alert_title'),
      text: i18n.t('error_alert_text'),
      reverseButtons: true,
      confirmButtonText: i18n.t('ok'),
      cancelButtonText: i18n.t('cancel'),
    })
  }

  if (status === 401 && store.getters['auth/check']) {
    Swal.fire({
      type: 'warning',
      title: i18n.t('token_expired_alert_title'),
      text: i18n.t('token_expired_alert_text'),
      reverseButtons: true,
      confirmButtonText: i18n.t('ok'),
      cancelButtonText: i18n.t('cancel'),
    }).then(() => {
      store.commit('auth/LOGOUT')

      router.push({ name: 'login' })
    })
  }

  return Promise.reject(error)
})

Vue.use({
  install (Vue) {
    Vue.prototype.$http = instance
  },
})

export default instance
