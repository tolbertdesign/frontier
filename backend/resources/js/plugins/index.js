import 'nprogress/nprogress.css'
// import '@fortawesome/fontawesome-free/css/all.min.css'

import '@/plugins/buefy'
import '@/plugins/fontawesome'
import '@/plugins/gravatar'
import '@/plugins/tooltips'
import '@/plugins/vue-croppa'
import '@/plugins/vue-meta'
import '@/plugins/vuelidate'

// import '@/plugins/validation'
