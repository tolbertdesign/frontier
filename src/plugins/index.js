import '@/plugins/gravatar'
import '@/plugins/vue-particles'
import '@/plugins/portal-vue'
import '@/plugins/tooltips'
// import '@/plugins/validation'
