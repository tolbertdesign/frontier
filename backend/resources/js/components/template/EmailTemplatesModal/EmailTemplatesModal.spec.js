import { shallowMount, createLocalVue } from '@vue/test-utils'
import EmailTemplatesModal from './EmailTemplatesModal'
import LanguageParser from '@/mixins/LanguageParser'
import Buefy from 'buefy'
import '@/plugins/fontawesome'

const localVue = createLocalVue()
localVue.use(Buefy, {
  defaultIconPack: 'fas',
})

localVue.mixin(LanguageParser)

describe('EmailTemplatesModal', () => {
  const $route = {
    params: {
      participantUserId: 2,
    },
  }
  const user = {
    first_name: 'Martha',
    programs: [
      {
        id: 1,
        event_name: 'Event Name',
        microsite: {
          funds_raised_for: 'Funds are needed for technology',
        },
        unit: {
          name: 'lap',
          name_plural: 'laps',
          modifier: 'per',

        },
        participants: [
          {
            id: 2,
            user_id: 3,
            first_name: 'Megan',
            profile: {
              pledge_goal: 100,
            },
            special_urls: [
              {
                id: 1,
                user_id: 3,
                referrer: {
                  id: 6,
                },
                short_key: 'AbCdEf',
              },
              {
                id: 2,
                user_id: 3,
                referrer: {
                  id: 7,
                },
                short_key: 'gHiJkL',
              },
              {
                id: 3,
                user_id: 3,
                referrer: {
                  id: 8,
                },
                short_key: 'mNoPqR',
              },
            ],
          },
        ],
      },
    ],
  }

  const participant2 = {
    id: 13,
    user_id: 68,
    first_name: 'Sallie',
    profile: {
      pledge_goal: 10,
    },
    special_urls: [
      {
        id: 5,
        user_id: 68,
        referrer: {
          id: 6,
        },
        short_key: 'IUYjhg',
      },
      {
        id: 6,
        user_id: 68,
        referrer: {
          id: 7,
        },
        short_key: 'jhYUGjb',
      },
      {
        id: 7,
        user_id: 68,
        referrer: {
          id: 8,
        },
        short_key: 'WEkJHGg',
      },
    ],
  }

  const participant3 = {
    id: 40,
    user_id: 80,
    first_name: 'Mary',
    profile: {
      pledge_goal: 15,
    },
    special_urls: [
      {
        id: 9,
        user_id: 80,
        referrer: {
          id: 6,
        },
        short_key: 'artsyer',
      },
      {
        id: 20,
        user_id: 80,
        referrer: {
          id: 7,
        },
        short_key: 'sgfKHdd',
      },
      {
        id: 30,
        user_id: 80,
        referrer: {
          id: 8,
        },
        short_key: 'ppPjbvs',
      },
    ],
  }

  const $store = {
    state: {
      User: user,
      lang: {
        emails_we_send: 'Emails We Send',
        sponsor_name: 'Sponsor Name',
        template: 'Coming up, :names :is_are participating in the :event_name and :is_are helping raise funds for :funds_raised_for. :names will complete 30-35 :plural_unit_name in the :event_name as a $:pledge_goal :unit_modifier :unit_name pledge goal. Will you help us?',
        click_here_reach_for_greatness: 'Click here to support participants',
        thanks_so_much: 'Thank you',
      },
    },
  }

  it('is a Vue instance', () => {
    const wrapper = shallowMount(EmailTemplatesModal, {
      localVue,
      mocks: { $store, $route },

    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('it renders correctly with 1 participant', () => {
    const wrapper = shallowMount(EmailTemplatesModal, {
      localVue,
      mocks: { $store, $route },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it renders correctly with 2 participants', () => {
    $store.state.User.programs[0].participants.push(participant2)
    const wrapper = shallowMount(EmailTemplatesModal, {
      localVue,
      mocks: { $store, $route },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it renders correctly with 3 or more participants', () => {
    $store.state.User.programs[0].participants.push(participant3)
    const wrapper = shallowMount(EmailTemplatesModal, {
      localVue,
      mocks: { $store, $route },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
