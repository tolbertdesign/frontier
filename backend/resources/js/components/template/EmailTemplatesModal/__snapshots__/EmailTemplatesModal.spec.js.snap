// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailTemplatesModal it renders correctly with 1 participant 1`] = `
<accordionmodal-stub header="Emails We Send" content="" footer="" class="mx-4">
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan is participating in the Event Name and is helping raise funds for Funds are needed for technology. Megan will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/AbCdEf">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub>
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan is participating in the Event Name and is helping raise funds for Funds are needed for technology. Megan will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/gHiJkL">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub>
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan is participating in the Event Name and is helping raise funds for Funds are needed for technology. Megan will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/mNoPqR">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub> <template></template>
</accordionmodal-stub>
`;

exports[`EmailTemplatesModal it renders correctly with 2 participants 1`] = `
<accordionmodal-stub header="Emails We Send" content="" footer="" class="mx-4">
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan &amp; Sallie are participating in the Event Name and are helping raise funds for Funds are needed for technology. Megan &amp; Sallie will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/AbCdEf">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub>
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan &amp; Sallie are participating in the Event Name and are helping raise funds for Funds are needed for technology. Megan &amp; Sallie will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/gHiJkL">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub>
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan &amp; Sallie are participating in the Event Name and are helping raise funds for Funds are needed for technology. Megan &amp; Sallie will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/mNoPqR">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub> <template></template>
</accordionmodal-stub>
`;

exports[`EmailTemplatesModal it renders correctly with 3 or more participants 1`] = `
<accordionmodal-stub header="Emails We Send" content="" footer="" class="mx-4">
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan, Sallie, and Mary are participating in the Event Name and are helping raise funds for Funds are needed for technology. Megan, Sallie, and Mary will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/AbCdEf">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub>
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan, Sallie, and Mary are participating in the Event Name and are helping raise funds for Funds are needed for technology. Megan, Sallie, and Mary will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/gHiJkL">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub>
  <accordion-stub title="" content="" animation="zoom-out"><template></template>
    <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
      <p class="mb-4">Sponsor Name,</p>
      <p class="mb-4">
        Coming up, Megan, Sallie, and Mary are participating in the Event Name and are helping raise funds for Funds are needed for technology. Megan, Sallie, and Mary will complete 30-35 laps in the Event Name as a $100 per lap pledge goal. Will you help us?
      </p>
      <p class="mb-4"><a href="/a/s/mNoPqR">
          Click here to support participants
        </a></p>
      <p class="mb-4">
        Thank you
      </p>
      <p class="mb-4">- Martha</p>
    </div>
  </accordion-stub> <template></template>
</accordionmodal-stub>
`;
